packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -57,63 +58,79 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
     );
   }
 
-  // Do not access this field directly; use [_httpClient] instead.
-  // We set `autoUncompress` to false to ensure that we can trust the value of
-  // the `Content-Length` HTTP header. We automatically uncompress the content
-  // in our call to [consolidateHttpClientResponseBytes].
-  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
+  static Future<Isolate> _pendingLoader;
+  static List<Function> _pendingLoaderUsers;
+  static SendPort _requestPort;
 
-  static HttpClient get _httpClient {
-    HttpClient client = _sharedHttpClient;
-    assert(() {
-      if (debugNetworkImageHttpClientProvider != null)
-        client = debugNetworkImageHttpClientProvider();
-      return true;
-    }());
-    return client;
-  }
-
-  Future<ui.Codec> _loadAsync(
+  Future<ui.Codec> _loadAsyncOnDesignatedIsolate(
     NetworkImage key,
     StreamController<ImageChunkEvent> chunkEvents,
   ) async {
     try {
       assert(key == this);
 
       final Uri resolved = Uri.base.resolve(key.url);
-      final HttpClientRequest request = await _httpClient.getUrl(resolved);
-      headers?.forEach((String name, String value) {
-        request.headers.add(name, value);
+
+      final Completer<TransferableTypedData> completer = Completer<TransferableTypedData>();
+      final RawReceivePort port = RawReceivePort((dynamic message) {
+        if (message is TransferableTypedData) {
+          completer.complete(message);
+        } else if (message is ImageChunkEvent) {
+          chunkEvents.add(message);
+        } else {
+          completer.completeError(message);
+        }
       });
-      final HttpClientResponse response = await request.close();
-      if (response.statusCode != HttpStatus.ok)
-        throw Exception('HTTP request failed, statusCode: ${response?.statusCode}, $resolved');
-
-      final Uint8List bytes = await consolidateHttpClientResponseBytes(
-        response,
-        onBytesReceived: (int cumulative, int total) {
-          chunkEvents.add(ImageChunkEvent(
-            cumulativeBytesLoaded: cumulative,
-            expectedTotalBytes: total,
-          ));
-        },
-      );
-      if (bytes.lengthInBytes == 0)
+
+      final _DownloadRequest downloadRequest = _DownloadRequest(port.sendPort,
+          resolved, headers, debugNetworkImageHttpClientProvider);
+      if (_requestPort != null) {
+        _requestPort.send(downloadRequest);
+      } else {
+        if (_pendingLoader == null) {
+          _pendingLoaderUsers = <Function>[];
+          _pendingLoader = _setupIsolate(chunkEvents, completer);
```

Comment:
Don't pass `chunkEvents` to `_setupIsolate()` (it's not used, and if it were used, that'd be a bug since it's a shared isolate).

Reply:
Done.